src/core/sagas/products.ts

```diff
@@ -100,12 +100,19 @@ export namespace Tasks {
       const state: Store.State = yield effects.select();
       const products = Selectors.productsWithMetadata(state);
       const pageSize = action.payload.amount;
+      const recordCount = Selectors.recordCount(state);
 
       let skip;
       if (action.payload.forward) {
-        skip = products[products.length - 1].index;
+        skip = products.length === 0 ? 0 : products[products.length - 1].index;
```

Comment:
Alternatively, `products.length && products[products.length - 1].index`, but I don't think it's as clear what the intent is.